mungegithub/mungers/milestone-maintainer_test.go

```diff
@@ -0,0 +1,233 @@
+// determined and applied to an issue.  Comprehensive testing is left
+// to TestNotificationState.
+//
+// TODO(marun) Enable testing of comment removal
+func TestMilestoneMaintainer(t *testing.T) {
+	runtime.GOMAXPROCS(runtime.NumCPU())
+
+	activeMilestone := "v1.8"
+	milestone := &githubapi.Milestone{Title: &activeMilestone, Number: intPtr(1)}
+	m := MilestoneMaintainer{activeMilestone: activeMilestone}
+
+	issue := github_test.Issue("user", 1, []string{}, false)
+	issue.Milestone = milestone
+
+	config := &github.Config{Org: "o", Project: "r"}
+	client, server, mux := github_test.InitServer(t, issue, nil, nil, nil, nil, nil, nil)
+	config.SetClient(client)
+
+	path := fmt.Sprintf("/repos/%s/%s/issues/%d", config.Org, config.Project, *issue.Number)
+
+	mux.HandleFunc(fmt.Sprintf("%s/labels", path), func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		out := []githubapi.Label{{}}
+		data, err := json.Marshal(out)
+		if err != nil {
+			t.Errorf("Unexpected error: %v", err)
+		}
+		w.Write(data)
+	})
+
+	var comments []githubapi.IssueComment
+	mux.HandleFunc(fmt.Sprintf("%s/comments", path), func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "POST" {
+			c := new(githubapi.IssueComment)
+			json.NewDecoder(r.Body).Decode(c)
+			comments = append(comments, *c)
+			w.WriteHeader(http.StatusOK)
+			data, err := json.Marshal(githubapi.IssueComment{})
+			if err != nil {
+				t.Errorf("Unexpected error: %v", err)
+			}
+			w.Write(data)
+			return
+		}
+		if r.Method == "GET" {
+			w.WriteHeader(http.StatusOK)
+			data, err := json.Marshal([]githubapi.IssueComment{})
+			if err != nil {
+				t.Errorf("Unexpected error: %v", err)
+			}
+			w.Write(data)
+			return
+		}
+		t.Fatalf("Unexpected method: %s", r.Method)
+	})
+
+	obj, err := config.GetObject(*issue.Number)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	m.Munge(obj)
+
+	expectedLabel := milestoneLabelsIncompleteLabel
+	if !obj.HasLabel(expectedLabel) {
+		t.Fatalf("Issue labels do not include '%s'", expectedLabel)
+	}
+
+	if len(comments) != 1 {
+		t.Fatalf("Expected comment count of %d, got %d", 1, len(comments))
+	}
+
+	server.Close()
+}
+
+func milestoneTestComment(label string, createdAt time.Time) *c.Comment {
+	n := &c.Notification{
+		Name:      milestoneNotifierName,
+		Arguments: label,
+		Context:   "foo",
+	}
+	return &c.Comment{
+		Body:      stringPtr(n.String()),
+		CreatedAt: &createdAt,
+	}
+}
+
+func milestoneLabelEvents(label string, createdAt time.Time) []*githubapi.IssueEvent {
+	return []*githubapi.IssueEvent{
+		{
+			Event: stringPtr("labeled"),
+			Label: &githubapi.Label{
+				Name: &label,
+			},
+			CreatedAt: &createdAt,
+			Actor: &githubapi.User{
+				Login: stringPtr(milestoneTestBotName),
+			},
+		},
+	}
+}
+
+// TestNotificationState validates how the notificationState() method
+// responds to the possible label and notification states of issues in
+// a milestone.
+func TestNotificationState(t *testing.T) {
```

Comment:
Do we have any tests for the actual contents of the realized templates?  If not, it would be helpful to prevent regressions to the templates.